### PR TITLE
pull in PWB Code Server upstream changes and use Positron product name

### PR DESCRIPTION
**Pull request by melissa-barca**
Pulls in changes from https://github.com/rstudio/vscode-server/pull/142 and updates the product name to be Positron-specific. 

This will allow the Workbench team to cleanly implement the RPCs necessary to save session state between user logouts. 

### QA Notes

Some work will still need to be done on the Workbench side. See https://github.com/rstudio/rstudio-pro/issues/6820 
The Workbench team will need to retest the VS Code state work and rework those existing tests to work with Positron.

### CODE REVIEW

File: test/automation/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "vscode-automation",
-  "version": "1.71.0",
```
Comment (sharon-wang) [inline test/automation/package.json:3]: Was this change intentional? It looks like this version hasn't been bumped upstream in vscode: https://github.com/microsoft/vscode/blob/main/test/automation/package.json

**Review by sharon-wang — COMMENTED** [review]
Could we also bring in the Workbench extension version bump to `product.json`?

File: test/automation/package.json (hunk 1 above)
Comment (melissa-barca) [reply test/automation/package.json]: Nope, not sure where that came from. Thank you for catching it!

**Review by sharon-wang — APPROVED** [review]
LGTM!

I'll cherry-pick this into positron `main`, since we probably won't be pulling in code server `main` until we do the 1.95 merge.

Related issue: https://github.com/posit-dev/positron/issues/5282
